templates/puppetserver-ingress-masters.yaml

```diff
@@ -16,18 +19,31 @@ metadata:
     {{- end }}
   name: {{ template "puppetserver.fullname" . }}-masters
 spec:
+  {{- if $apiIsStable }}
+  {{- if .Values.puppetserver.masters.ingress.ingressClassName }}
+  ingressClassName: {{ .Values.puppetserver.masters.ingress.ingressClassName }}
+  {{- end }}
+  {{- end }}
   rules:
     {{- range .Values.puppetserver.masters.ingress.hosts }}
     {{- $url := splitList "/" . }}
     - host: {{ first $url }}
       http:
         paths:
           - path: /{{ rest $url | join "/" }}
+            {{- if and $pathType $ingressSupportsPathType }}
+            pathType: {{ $pathType }}
+            {{- end }}
             backend:
+              {{- if $apiIsStable }}
               service:
                 name: {{ $serviceName }}
                 port:
                   number: {{ $servicePort }}
+              {{- else }}
+              serviceName: {{ $serviceName }}
+              servicePort: {{ $servicePort }}
+    {{- end }}
```

Comment:
And here.

Reply:
@MikaelSmith Fixed - though didn't affect anything.